uitests/tests/org.jboss.tools.fuse.ui.bot.tests/src/org/jboss/tools/fuse/ui/bot/tests/CamelCatalogDefaultTest.java

```diff
@@ -0,0 +1,126 @@
+import org.jboss.tools.fuse.reddeer.requirement.CamelCatalogRequirement;
+import org.jboss.tools.fuse.reddeer.requirement.CamelCatalogRequirement.CamelCatalog;
+import org.jboss.tools.fuse.reddeer.view.PaletteViewExt;
+import org.jboss.tools.fuse.ui.bot.tests.utils.ProjectFactory;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * Verifies components from Camel editor palette view against appropriate Camel catalog
+ * 
+ * @author djelinek
+ */
+@CamelCatalog
+@CleanWorkspace
+@CleanErrorLog
+@RunWith(RedDeerSuite.class)
+@OpenPerspective(FuseIntegrationPerspective.class)
+public class CamelCatalogDefaultTest {
+
+	private static final String PROJECT_NAME = "cbr";
+
+	private static final String CONTEXT = "camel-context.xml";
+
+	private static CamelCatalogUtils catalog;
+
+	@InjectRequirement
+	private static CamelCatalogRequirement catalogRequirement;
+
+	@BeforeClass
+	public static void setupTestEnvironment() {
+		new WorkbenchShell().maximize();
+		ConsolePreferenceUtil.setConsoleOpenOnError(false);
+		ConsolePreferenceUtil.setConsoleOpenOnOutput(false);
+		ProjectFactory.newProject(PROJECT_NAME).template(ProjectTemplate.CBR).type(ProjectType.SPRING)
+				.version(catalogRequirement.getConfiguration().getVersion()).create();
+		catalog = new CamelCatalogUtils(catalogRequirement.getConfiguration().getHome());
+	}
+
+	@AfterClass
+	public static void cleanSetup() {
+		EditorHandler.getInstance().closeAll(true);
+		ProjectFactory.deleteAllProjects();
+	}
+
+	/**
+	 * <p>
+	 * Test for verifying presence of components against appropriate Camel catalog
+	 * </p>
+	 * <ol>
+	 * <li>Create a new Fuse Integration Project - template: CBR, project type: Spring</li>
+	 * <li>Get list of components from Camel catalog</li>
+	 * <li>Get list of components from Camel Editor Palette view</li>
+	 * <li>Verify that all components are available in catalog</li>
+	 * </ol>
+	 */
+	@Test
+	public void testComponentPresence() {
+		new CamelEditor(CONTEXT).activate();
+		new PaletteView().open();
+		List<String> missing = new ArrayList<>();
+		for (String component : new PaletteViewExt().getGroupTools("Components")) {
+			if (!isExist(component)) {
+				missing.add(component);
+			}
+		}
+		if (!missing.isEmpty()) {
+			StringBuilder builder = new StringBuilder();
+			builder.append("List of missing components:");
+			for (String m : missing) {
+				builder.append("\n").append(m);
+			}
+			fail(builder.toString());
+		}
+	}
+
+	private boolean isExist(String name) {
+		/**
+		 * 'Generic' - general component (it is not route component), without properties
+		 * 'ActiveMQ' - it is not part of the standard Apache Camel distribution, it can be used to extend Camel's functionality
+		 * 'Process' - not standard component, it is interface used to implement consumers of message exchanges 
+		 */
+		List<String> skip = new ArrayList<>();
```

Comment:
This list should be extracted as a class field. You will avoid creating this array every time when the method is called ;-).